docs/data/date-pickers/fields/fields.md

```diff
@@ -37,25 +37,29 @@ The single input range fields are not supported on the range pickers yet (you ca
 
 The field components have an internal state to update the visible value.
 
-It will only call the `onChange` callback when all the sections of the modified date are filled.
+It will only call the `onChange` callback when:
 
-#### On single element fields
+- all the sections of the modified date are filled
+- all the sections of the modified date are empty (will return `null` for the modified date)
+- all the sections of the modified date were filled, and the user cleaned the content of one section (will return `Invalid date` for modified date)
 
-On a single date field (`DateField` / `DateTimeField` / `TimeField`),
-`onChange` will be called if all the sections are filled.
+#### On simple fields
 
-In the example below, `onChange` will not be fired until the date is fully completed:
+On simple fields (`DateField` / `DateTimeField` / `TimeField`),
+`onChange` will be called if the date matches one of the condition above.
+
+In the example below, `onChange` will be called once when filling the first section and once when filling the last one:
 
 {{"demo": "LifeCycleDateFieldEmpty.js", "defaultCodeOpen": false}}
 
 #### On range fields [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)
 
-On a date range field (`SingleInputDateRangeField` / `MultiInputDateRangeField`),
-`onChange` will be called if all the sections of the date you modified are filled,
-even if some sections of the other date are not filled.
+On range fields (`SingleInputDateRangeField` / `MultiInputDateRangeField` / ... ),
+`onChange` will be called if the date you are modifying is matching one of the condition above,
```

Comment:
```suggestion
`onChange` will be called if any of the dates you are modifying matches one of the conditions above,
```

Reply:
That's not true
If you fill your start date, you have one of the date that matches one of the condition above.
And yet, when editing the end date, you won't have an `onChange` call when filling the 2nd section.

The rule is applied on the modified date, whatever value the other date might have.

Reply:
Got it!